pdf_compression/models/ir_attachment.py

```diff
@@ -0,0 +1,171 @@
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                    ])
+                elif pdfa_option == 'pdfa2b':
+                    local_command_args.extend([
+                        '-dPDFA=2',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                        '-dEmbedAllFonts=true',
+                        '-dMaxSubsetPct=100',
+                        '-dSubsetFonts=true'
+                    ])
+
+            if report_override != 'default':
+                if report_override == 'pdfa1b':
+                    local_command_args.extend([
+                        '-dPDFA=1',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                    ])
+                elif report_override == 'pdfa2b':
+                    local_command_args.extend([
+                        '-dPDFA=2',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                        '-dEmbedAllFonts=true',
+                        '-dMaxSubsetPct=100',
+                        '-dSubsetFonts=true'])
+
+            if compression == 'no' and (pdfa_option == 'pdfa1b' or
+                                        report_override == 'pdfa1b'):
+                local_command_args.extend(['-dPDFSETTINGS=/default'])
+            elif compression == 'prepress':
+                local_command_args.extend(['-dPDFSETTINGS=/prepress'])
+            elif compression == 'screen':
+                local_command_args.extend(['-dPDFSETTINGS=/screen'])
+            elif compression == 'ebook':
+                local_command_args.extend(['-dPDFSETTINGS=/ebook'])
+            elif compression == 'printer':
+                local_command_args.extend(['-dPDFSETTINGS=/printer'])
+            elif compression == 'default':
+                local_command_args.extend(['-dPDFSETTINGS=/default'])
+
+            local_command_args.extend(
+                ['-sDEVICE=pdfwrite', '-dCompatibilityLevel=1.4',
+                 '-dNOPAUSE', '-dQUIET', '-dBATCH', '-dNOOUTERSAVE',
+                 '-sOutputFile=' + output_file, input_file])
+
+            try:
+                ghostscript = [_get_ghostscript_bin()] + command_args + \
+                    local_command_args
+                process = subprocess.Popen(ghostscript, stdout=subprocess.PIPE,
+                                           stderr=subprocess.PIPE)
+                out, err = process.communicate()
+
+                if err:
+                    _logger.info('ghostscript failed . Message: %s' % err)
+
+                with open(output_file, 'rb') as f3:
+                    pdf = base64.encodestring(f3.read())
+
+                os.unlink(input_file)
+                os.unlink(output_file)
+
+            except (OSError, IOError, ValueError):
```

Comment:
IMO the try/except:raise is unnecessary unless we're doing something before the raise.